src/components/worker/worker.py

```diff
@@ -109,65 +109,204 @@ def __init__(
+                mono_cam.setResolution(resolution)
+                mono_cam.setBoardSocket(sensor.socket)
+                mono_cam.setFps(sensor.frame_rate)
+                return mono_cam
+
+            stereo_pair = self.cfg.sensors.stereo_pair
+            depth = None
+            if stereo_pair:
+                LOGGER.debug("Creating stereo depth component.")
+                mono1, mono2 = stereo_pair
+                mono_cam_1 = make_mono_cam(mono1)
+                mono_cam_2 = make_mono_cam(mono2)
+
+                depth = self.pipeline.create(dai.node.StereoDepth)
+                depth.setDefaultProfilePreset(
+                    dai.node.StereoDepth.PresetMode.HIGH_DENSITY
+                )
+                depth.initialConfig.setMedianFilter(dai.MedianFilter.KERNEL_7x7)
+
+                mono_cam_1.out.link(depth.left)
+                mono_cam_2.out.link(depth.right)
+
+                xout_depth = self.pipeline.create(dai.node.XLinkOut)
+                xout_depth.setStreamName(self.depth_stream_name)
+                depth.disparity.link(xout_depth.input)
+
+                color_sensors = self.cfg.sensors.color_sensors
+                if color_sensors and len(color_sensors) > 0:
+                    sensor_color_align = color_sensors[0]
+                    depth.setDepthAlign(sensor_color_align.socket)
+            return depth
+
+        def configure_pc(
+            color_node: Optional[dai.node.ColorCamera],
+            depth_node: Optional[dai.node.StereoDepth],
+        ) -> None:
+            """
+            Creates and configures point clouds— or doesn't
+            (based on the config)
+            """
+            if depth_node and self.user_wants_pc:
+                pointcloud = self.pipeline.create(dai.node.PointCloud)
+                sync = self.pipeline.create(dai.node.Sync)
+                xOut = self.pipeline.create(dai.node.XLinkOut)
+
+                xOut.setStreamName(self.pc_stream_name)
+                xOut.input.setBlocking(False)
+
+                # Link the nodes
+                depth_node.depth.link(pointcloud.inputDepth)
+                if color_node:
+                    color_node.isp.link(sync.inputs["rgb"])
+                pointcloud.outputPointCloud.link(sync.inputs["pcl"])
+                sync.out.link(xOut.input)
+
+        self.pipeline = dai.Pipeline()
+        try:
+            stage = "color"
+            color_node = configure_color()
+
+            stage = "stereo"
+            depth_node = configure_stereo()
 
-        if (
-            not self.should_exec
-        ):  # possible SIGTERM between initializing OakCamera and here
-            LOGGER.debug("Stopping configuration due to termination signal")
-            return
+            stage = "point cloud"
+            configure_pc(color_node, depth_node)
 
-        self._config_oak_camera()
+            LOGGER.info("Successfully configured pipeline.")
+        except Exception as e:
```

Comment:
I wish we could do a better job of catching the error type, but since it comes from C, we lose its type